pymatgen/core/structure.py

```diff
@@ -341,7 +341,9 @@ def charge(self) -> float:
         charge = 0
         for site in self:
             for specie, amt in site.species.items():
-                charge += getattr(specie, "oxi_state", 0) * amt
+                if hasattr(specie,"oxi_state") and specie.oxi_state:
```

Comment:
Maybe less readable? Shorter in any case.

```suggestion
                charge += (getattr(specie, "oxi_state", 0) or 0)* amt
```

If we go with `hasattr`, I'd say let's be explicit:

```diff
- if hasattr(specie,"oxi_state") and specie.oxi_state:
+ if hasattr(specie,"oxi_state") and specie.oxi_state != 0:
```

Reply:
The first one works correctly. It takes 0 in case specie.oxi_state = None.
The second does not work as it is. It should be specie.oxi_state != None.

The first one is less clear, while the second more explicit.
I have no preference ;-)

Reply:
Pls just use getattr. Using hasattr is very verbose.